utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -15,67 +15,189 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! `Structopt`-ready structs for `try-runtime`.
-
-use parity_scale_codec::{Decode, Encode};
-use remote_externalities::{rpc_api, Builder, Mode, OfflineConfig, OnlineConfig, SnapshotConfig};
+//! # Try-runtime
+//!
+//! Substrate's ultimate testing framework for the power users.
+//!
+//! > As the name suggests, `try-runtime` is a detailed testing framework that gives you a lot of
+//! control over what is being executed in which environment. It is recommended that user's first
+//! familiarize themselves with substrate in depth, particularly the execution model. It is critical
+//! to deeply understand how the wasm/native interactions, and the runtime apis work in the
+//! substrate runtime.
+//!
+//! #### Resources
+//!
+//! Some resources about the above:
+//!
+//! 1. https://substrate.dev/docs/en/knowledgebase/integrate/try-runtime
+//! 2. https://www.crowdcast.io/e/substrate-seminar/41
+//! 3. https://substrate.dev/docs/en/knowledgebase/advanced/executor
+//!
+//! ---
+//!
+//! ## Overview
+//!
+//! The basis of all try-runtime commands is the same: connect to a live node, scrape its *state*
+//! and put it inside a `TestExternalities`, then call into a *specific runtime-api* using the given
+//! state and some *runtime*.
+//!
+//! All of the variables in the above statement are made *italic*. Let's look at each of them:
+//!
+//! 1. **State** is the key-value pairs of data that consist the important information that any
+//! blockchain is keeping. A state can be full (all key-value pairs), or be partial (only pairs
+//! related to some pallets). Moreover, some keys are special and live outside of pallets, known as
```

Comment:
```suggestion
//! related to some pallets). Moreover, some keys are special and are not related to specific pallets, known as
```

Just trying to subtley seperate the conceptual layers of logic vs storage. Keys may belong to a pallet, but "live" in a pallet does not make sense to me (feel free to ignore bc this is very opionated)